### ISSUE TITLE
Should `bq` be a pre-built builder?

### ISSUE THREAD

**Issue by polleyg** [GoogleCloudPlatform/cloud-builders-community#94]
_Apologies if using an issue to ask this is the wrong place._

Would it be worth considering `bq` as a [pre-built builder](https://github.com/GoogleCloudPlatform/cloud-builders)? This is the case for the similar `gsutil` builder, and it feels to me like `bq` should have the same privileges too.

BigQuery is a native GCP tool, and it's the core tool to use on GCP for data and analytics. So having it as a pre-built image might make more sense.

**Comment by ImJasonH** [comment]
The main reason we haven't promoted it is because we haven't seen much usage of it in GCB, but I admit there's a bit of a chicken-and-egg problem with that logic.

Usually we only pre-build images we expect to have significant usage, and usually to which someone has committed to keeping up-to-date. Since `bq` is a GCP tool and is kept up to date by `gcloud`, that's less of a burden at least.

**Comment by polleyg** [comment]
Fair enough! That said, it was the first one I looked for when I started poking around with GCB, but that's because we focus heavily on BQ where I work.

Might still be worth promoting it due to the fact it's a native GCP tool like you said. Feels inconsistent that `gsutil` is a first class citizen, but `bq` is not.

**Comment by ImJasonH** [comment]
Can you help me understand how you'd use `bq` in your cloud build process? `gsutil` has a fairly obvious use case, but I'm not sure about `bq`.

Do you plan to run queries or insert records (or both) during your build?

**Comment by polleyg** [comment]
There's a few uses cases that I can think of off the top of my head:

1. Running SQL
2. Creating datasets & table definitions
3. Creating & updating views
4. Patching datasets & tables
5. Copying tables/datasets (e.g. backing up)

**Comment by ImJasonH** [comment]
Are these theoretical use cases you might be interested in, or concrete real-world workflows you're doing today, which would be smoother if the builder was prebuilt and managed by the GCB team?

Looking at our logs it looks like nobody has actually used the `bq` community builder in the last 3 weeks, so I think I'd want to see some more actual usage before committing to building/maintaining the `bq` builder in perpetuity.

**Comment by polleyg** [comment]
These are real-world, production use cases. Currently we deploy all our pipelines (which use BQ) through a combination of the `bq` cli tool and API calls. It's a little bit messy/hacky and the service account management (IAM) can be hard too. 

Yes, this would make it a lot smoother :)

**Comment by ImJasonH** [comment]
I think my question can be more clearly stated as, "are you using the `bq` builder today, and experiencing pain because it's not prebuilt?" Evidence suggests that ~nobody is using the `bq` community builder today, though it's possible I got my query wrong, or that you're building it as something other than `bq`.

Another option in the meantime would be to just use the `gcloud` builder and override the `entrypoint` -- that's all the `bq` builder would be doing anyway AFAICT. That image is prebuilt (it's also _huge,_ so thankfully it's cached!), and should work just fine for now, though it's a bit uglier than it could be:

```
steps:
- name: gcr.io/cloud-builders/gcloud
  entrypoint: bq
  args: ['load', '...']
```

If that unblocks you for now, great, otherwise let me know and I'll fix whatever's broken.
